src/opm/io/eclipse/EInit.cpp

```diff
@@ -0,0 +1,161 @@
+
+    lgrname = "global";
+
+    for (size_t n = 0; n < array_name.size(); n++) {
+        if (array_name[n] == "LGR"){
+            auto lgr = this->get<std::string>(n);
+            lgrname = lgr[0];
+
+            if (std::find(lgr_names.begin(), lgr_names.end(), lgrname) == lgr_names.end()){
+                lgr_names.push_back(lgrname);
+                lgr_array_index.push_back({});
+                lgr_nijk.push_back({});
+                lgr_nactive.push_back(0);
+            }
+        }
+
+        if (array_name[n] == "LGRSGONE")
+            lgrname = "global";
+
+        if ((lgrname == "global") && (array_name[n] != "LGRSGONE"))
+            global_array_index[array_name[n]] = n;
+
+        if ((lgrname != "global") && (array_name[n] != "LGRSGONE") && (array_name[n] != "LGR"))
+        {
+           auto it = std::find(lgr_names.begin(), lgr_names.end(), lgrname);
+           size_t ind = std::distance(lgr_names.begin(), it);
+           lgr_array_index[ind][array_name[n]] = n;
+        }
+
+        if (array_name[n] == "INTEHEAD")
+        {
+            auto inteh = getImpl(n, INTE, inte_array, "integer");
+
+            if (lgrname == "global") {
+                global_nijk = {inteh[8], inteh[9], inteh[10]};
+                global_nactive = inteh[11];
+            } else {
+               auto it = std::find(lgr_names.begin(), lgr_names.end(), lgrname);
+               size_t lgr_ind = std::distance(lgr_names.begin(), it);
+               lgr_nijk[lgr_ind] = {inteh[8], inteh[9], inteh[10]};
+               lgr_nactive[lgr_ind] = inteh[11];
+            }
+        }
+    }
+}
+
+
+int EInit::get_array_index(const std::string& name, const std::string& grid_name) const
+{
+    if (grid_name == "global"){
+        if (global_array_index.count(name) == 0)
+            OPM_THROW(std::runtime_error, "Map key not found in global_array_index");
+
+        return global_array_index.at(name);
+
+    } else {
+
+        int lgr_index = get_lgr_index(grid_name);
+
+        if (lgr_array_index[lgr_index].count(name) == 0)
+            OPM_THROW(std::invalid_argument, "Map key not found in lgr_array_index");
+
+        return lgr_array_index[lgr_index].at(name);
+    }
+};
+
+int EInit::activeCells(const std::string& grid_name) const
+{
+    if (grid_name == "global")
+        return global_nactive;
+    else
+        return lgr_nactive[get_lgr_index(grid_name)];
+}
+
+const std::array<int, 3>& EInit::grid_dimension(const std::string& grid_name) const
+{
+    if (grid_name == "global")
+        return global_nijk;
+    else
+        return lgr_nijk[get_lgr_index(grid_name)];
+}
+
+bool EInit::hasLGR(const std::string& name) const{
+    if (std::find(lgr_names.begin(), lgr_names.end(), name) == lgr_names.end())
+        return false;
+    else
+        return true;
+}
+
+int EInit::get_lgr_index(const std::string& grid_name) const
+{
+    if (std::find(lgr_names.begin(), lgr_names.end(), grid_name) == lgr_names.end()) {
```

Comment:
store the iterator, to avoid double search.

Reply:
Done